ios/ReactNativeCameraKit/RealCamera.swift

```diff
@@ -57,21 +71,64 @@ class RealCamera: NSObject, CameraProtocol, AVCaptureMetadataOutputObjectsDelega
                 self.removeObservers()
             }
         }
+        
+        motionManager?.stopAccelerometerUpdates()
+        
+        NotificationCenter.default.removeObserver(self, name: UIDevice.orientationDidChangeNotification, object: UIDevice.current)
+        
+        UIDevice.current.endGeneratingDeviceOrientationNotifications()
     }
 
     deinit {
         removeObservers()
     }
 
     // MARK: - Public
-
+    
+    func initializeMotionManager() {
+        motionManager = CMMotionManager()
+        motionManager?.accelerometerUpdateInterval = 0.2
+        motionManager?.gyroUpdateInterval = 0.2
+        motionManager?.startAccelerometerUpdates(to: (OperationQueue.current)!, withHandler: {
+            (accelerometerData, error) -> Void in
+            if error != nil {
+                print("\(error!)")
+            }
+            guard let acceleration = accelerometerData?.acceleration else {
+                print("no acceleration data")
+                return
+            }
+            var orientationNew: UIDeviceOrientation
+            if acceleration.x >= 0.75 {
+                orientationNew = .landscapeLeft
+            } else if acceleration.x <= -0.75 {
+                orientationNew = .landscapeRight
+            } else if acceleration.y <= -0.75 {
+                orientationNew = .portrait
+            } else if acceleration.y >= 0.75 {
+                orientationNew = .portraitUpsideDown
+            } else {
+                // Consider same as last time
+                return
+            }
+            
+            if orientationNew == self.deviceOrientation {
+                return
+            }
+            self.deviceOrientation = orientationNew
```

Comment:
1. `CMMotionManager` won't work in the simulator, so it needs to be in the specific implmenentation. The rotation you can do with the simulator does not cause "actual" motion.
2. I tried that. No, videoOrientation mustn't always be updated based on motion. Only after a _significant_ change in orientation, or only taking a picture. Otherwise, UI orientation and motion may counter-indicate, causing video to be landscape, while UI is portrait or vice versa, which is technically correct but confusing to the user.  
   Please keep it as is, as I spent a significant amount of time testing all the use cases. You need to consider UI portrait (locked + unlocked), UI landscape left+right (not recommended but iPad UIs may need it), and every 3D rotation of the phone as well (portrait, landscape left, landscape right, upside down, 45 degree angle (=meaning no change from past orientation). If you compare the native iOS camera, it has the same behavior - where UI elements indicate portrait, but the captured photo is landscape. And that is what you'd want it to.